From 386dd9dc4411f87981f91773c8eff220ef22be53 Mon Sep 17 00:00:00 2001
From: Marko Lindqvist <cazfi74@gmail.com>
Date: Sun, 16 Jun 2024 02:09:45 +0300
Subject: [PATCH 30/30] Tex: Initialize map topology

See RM #663

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
---
 ai/tex/texaiworld.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ai/tex/texaiworld.c b/ai/tex/texaiworld.c
index 26f9d63ea2..97d6c55e1f 100644
--- a/ai/tex/texaiworld.c
+++ b/ai/tex/texaiworld.c
@@ -85,6 +85,7 @@ void texai_world_close(void)
 void texai_map_init(void)
 {
   map_init(&(texai_world.map), TRUE);
+  map_init_topology(&(texai_world.map));
   map_allocate(&(texai_world.map));
 }
 
-- 
2.43.0

