From 8d719e330a4c48024f50026f3053d15c828b2dd3 Mon Sep 17 00:00:00 2001
From: Marko Lindqvist <cazfi74@gmail.com>
Date: Wed, 17 Apr 2024 20:51:26 +0300
Subject: [PATCH 50/50] savegame: Fix loading invalid resources on FoW map

See RM #463

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
---
 server/savegame/savegame3.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/server/savegame/savegame3.c b/server/savegame/savegame3.c
index de2fa885a3..875839dec0 100644
--- a/server/savegame/savegame3.c
+++ b/server/savegame/savegame3.c
@@ -7160,9 +7160,11 @@ static void sg_load_player_vision(struct loaddata *loading,
     struct player_tile *plrtile = map_get_player_tile(ptile, plr);
 
     extra_type_by_cause_iterate(EC_RESOURCE, pres) {
-      if (BV_ISSET(plrtile->extras, extra_number(pres))
-          && terrain_has_resource(plrtile->terrain, pres)) {
+      if (BV_ISSET(plrtile->extras, extra_number(pres))) {
         plrtile->resource = pres;
+        if (!terrain_has_resource(plrtile->terrain, pres)) {
+          BV_CLR(plrtile->extras, extra_number(pres));
+        }
       }
     } extra_type_by_cause_iterate_end;
   } whole_map_iterate_end;
-- 
2.43.0

