From 6d0a68373ffd38ece9be93ddc71f1393392dc73d Mon Sep 17 00:00:00 2001
From: Marko Lindqvist <cazfi74@gmail.com>
Date: Thu, 18 Apr 2024 19:01:25 +0300
Subject: [PATCH 54/54] Fix inconsistent city workers after vision loss

See RM #472

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
---
 server/maphand.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/server/maphand.c b/server/maphand.c
index e462ba0fe9..11e74e969e 100644
--- a/server/maphand.c
+++ b/server/maphand.c
@@ -1055,6 +1055,9 @@ void map_change_seen(struct player *pplayer,
     if (pcity != NULL && city_owner(pcity) == pplayer) {
       city_map_update_empty(pcity, ptile);
       pcity->specialists[DEFAULT_SPECIALIST]++;
+      if (pcity->server.needs_arrange == CNA_NOT) {
+        pcity->server.needs_arrange = CNA_NORMAL;
+      }
     }
 
     update_player_tile_last_seen(pplayer, ptile);
@@ -2522,6 +2525,19 @@ void vision_clear_sight(struct vision *vision)
   const v_radius_t vision_radius_sq = V_RADIUS(-1, -1, -1);
 
   vision_change_sight(vision, vision_radius_sq);
+
+  /* Owner of some city might have lost vision of a tile previously worked */
+  players_iterate(pplayer) {
+    city_list_iterate(pplayer->cities, pcity) {
+      /* We are not interested about CNA_BROADCAST_PENDING, as the vision loss has
+       * not set it, and whatever set it should take care of it. */
+      if (pcity->server.needs_arrange == CNA_NORMAL) {
+        city_refresh(pcity);
+        auto_arrange_workers(pcity);
+        pcity->server.needs_arrange = CNA_NOT;
+      }
+    } city_list_iterate_end;
+  } players_iterate_end;
 }
 
 /**********************************************************************//**
-- 
2.43.0

