From cbf458a47de83ee449b76688da19e15f624e98eb Mon Sep 17 00:00:00 2001
From: Marko Lindqvist <cazfi74@gmail.com>
Date: Sun, 12 May 2024 06:58:58 +0300
Subject: [PATCH 68/81] Meson: Make libfreeciv to depend on gitrev

See RM #597

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 1309a687ed..1f2b068354 100644
--- a/meson.build
+++ b/meson.build
@@ -1365,7 +1365,7 @@ common_lib = library('freeciv',
   'common/workertask.c',
   'common/worklist.c',
   include_directories : common_inc,
-  sources: [specenum, enum_targets['common'], pack_common,
+  sources: [gitrev, specenum, enum_targets['common'], pack_common,
             tolua_com_a, tolua_com_z, tolua_game, tolua_signal],
   link_whole: fc_deps,
   dependencies: [zlib_dep,
-- 
2.43.0

