From 6228415a558a8adf6f715abe2d91ed2fd145b79a Mon Sep 17 00:00:00 2001
From: Marko Lindqvist <cazfi74@gmail.com>
Date: Tue, 28 May 2024 00:11:05 +0300
Subject: [PATCH 74/74] Meson: Make fc_server to depend on verhdr

See RM #682

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 5938236ee7..7a3f0288e8 100644
--- a/meson.build
+++ b/meson.build
@@ -1503,7 +1503,7 @@ server_lib = static_library('fc_server',
   'server/unittools.c',
   'server/voting.c',
   include_directories: server_inc,
-  sources: [ pack_server,
+  sources: [ verhdr, pack_server,
              tolua.process('server/scripting/tolua_fcdb.pkg',
                            'server/scripting/tolua_server.pkg')],
   dependencies: lua_dep
-- 
2.43.0

