From 48ff62e75292457e78b3075edd5ea2a734aca342 Mon Sep 17 00:00:00 2001
From: Marko Lindqvist <cazfi74@gmail.com>
Date: Sun, 12 May 2024 05:45:20 +0300
Subject: [PATCH 77/77] Meson: Make pack_client to depend on verhdr

See RM #596

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
---
 meson.build | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 1e1542c8c9..1309a687ed 100644
--- a/meson.build
+++ b/meson.build
@@ -1150,7 +1150,8 @@ pack_client = custom_target('packets_client',
                                       '@INPUT@',
                                       '--client-h', '@OUTPUT0@',
                                       '--client-c', '@OUTPUT1@'] + gen_packets_args,
-                            depend_files: files('common/generate_packets.py'))
+                            depend_files: files('common/generate_packets.py'),
+                            depends: verhdr)
 
 gitrev = custom_target('gitrev', output: 'fc_gitrev_gen.h',
                        command: [sh_exe, files('bootstrap/generate_gitrev.sh'),
-- 
2.43.0

