From 79e30f320d9f8f299e47014b37d517fd063688c0 Mon Sep 17 00:00:00 2001
From: Marko Lindqvist <cazfi74@gmail.com>
Date: Tue, 21 May 2024 09:50:07 +0300
Subject: [PATCH 89/92] Meson: Make fc_client_common to depend on verhdr

See RM #596

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index cd05e91f2e..0d6b15bead 100644
--- a/meson.build
+++ b/meson.build
@@ -2169,7 +2169,7 @@ client_common = static_library('fc_client_common',
   'client/update_queue.c',
   'client/voteinfo.c',
   'client/zoom.c',
-  sources: [audio_sdl_src, pack_client,
+  sources: [verhdr, audio_sdl_src, pack_client,
             tolua.process('client/luascript/tolua_client.pkg')],
   include_directories: client_inc,
   link_with: [common_lib],
-- 
2.43.0

